karapace/backup/api.py

```diff
@@ -263,29 +262,53 @@ def _consumer(config: Config, topic: str) -> Iterator[KafkaConsumer]:
         yield consumer
 
 
-@contextlib.contextmanager
-def _enable_producer_callback_errors() -> Iterator[None]:
-    global_value = Future.error_on_callbacks
-    Future.error_on_callbacks = True
-    try:
-        yield None
-    finally:
-        Future.error_on_callbacks = global_value
+class BackupRestorationProducer:
+    def __init__(self, kafka_producer: KafkaProducer) -> None:
+        self.producer = kafka_producer
+        self.producer_exception: Exception | None = None
+
+    def _producer_error_callback(self, exception: Exception) -> None:
+        LOG.error("Producer error: %r", exception)
```

Comment:
```suggestion
        LOG.exception(exception)
```

issue: I think we might want to log the full exception with traceback and all here?

Reply:
It seems this one regressed.